### PR TITLE
Fix techniques with same uniform name but different parameter

### PR DESCRIPTION
**Pull request by lilleyse**
Fixes https://github.com/AnalyticalGraphicsInc/gltf-pipeline/issues/488

Had to map uniforms based on technique so that different techniques' uniforms would not override each other. Also fixed an issue where a program would get duplicated if two techniques referenced it.

### CODE REVIEW

**Comment by liukangkanglkk99** [comment]
Nice going! Thanks @lilleyse, I tested this and it does work.

File: lib/moveTechniquesToExtension.js [hunk 1]
```diff
@@ -60,27 +61,35 @@ function moveTechniquesToExtension(gltf) {
...
             });
 
-            const programLegacy = gltf.programs[techniqueLegacy.program];
-            const program = {
-                name: programLegacy.name,
-                fragmentShader: undefined,
-                vertexShader: undefined,
-                glExtensions: glExtensions
-            };
+            if (!defined(seenPrograms[techniqueLegacy.program])) {
```
Comment (likangning93) [inline lib/moveTechniquesToExtension.js]: Does this need to set `techniqueLegacy.program` in an `else` block?

Comment (lilleyse) [reply lib/moveTechniquesToExtension.js]: Thanks, fixed!

**Comment by lilleyse** [comment]
@likangning93 ready for another look

**Comment by likangning93** [comment]
:+1:

**Pull request merged**